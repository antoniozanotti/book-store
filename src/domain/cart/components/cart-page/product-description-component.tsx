import { ProductLinkComponent } from "@/domain/product/components/product-link-component";
import { getAuthorsByProduct } from "@/domain/product/helpers/get-authors-by-product";
import { ProductWithPageAndBookPrisma } from "@/domain/product/types/product-prisma";

export function ProductDescription({
  product,
}: {
  product: ProductWithPageAndBookPrisma;
}) {
  return (
    <div className="h-fit">
      <ProductLinkComponent product={product} className="line-clamp-2">
        <h2>{product.name}</h2>
      </ProductLinkComponent>
      <span className="line-clamp-2 block text-sm">
        by {getAuthorsByProduct(product)}
      </span>
      <span className="block text-sm my-2">ISBN: {product.isbn}</span>
    </div>
  );
}
