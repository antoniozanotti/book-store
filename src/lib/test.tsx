import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
