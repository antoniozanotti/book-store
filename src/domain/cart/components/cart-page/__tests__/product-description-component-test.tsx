import { render, screen } from "@/lib/test";
import { ProductDescription } from "../product-description-component";
import { productWithPagEandBookMock } from "@/domain/product/__tests__/product-with-page-and-book-mock";

describe("ProductDescription", () => {
  it("should render", async () => {
    const { container } = render(<ProductDescription product={productWithPagEandBookMock} />);
    expect(container).toMatchSnapshot();
  });
});