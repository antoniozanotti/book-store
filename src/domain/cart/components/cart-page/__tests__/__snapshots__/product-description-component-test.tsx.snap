// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDescription should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <div
      class="h-fit"
    >
      <a
        class="line-clamp-2"
        href="/product/the-four-agreements-a-practical-guide-to-personal-freedom"
        title="The Four Agreements: A Practical Guide to Personal Freedom by Don Miguel Ruiz"
      >
        <h2>
          The Four Agreements: A Practical Guide to Personal Freedom
        </h2>
      </a>
      <span
        class="line-clamp-2 block text-sm"
      >
        by 
        Don Miguel Ruiz
      </span>
      <span
        class="block text-sm my-2"
      >
        ISBN: 
        978-1-315-92574-5
      </span>
    </div>
  </div>
</div>
`;
